### ISSUE TITLE
Unable to install template in dev store using provided instructions

### ISSUE THREAD

**Issue by MidwestInfoDev** [Shopify/shopify-app-template-php#413]
# Issue summary
After multiple attempts I have been unable to successfully install an app using this template in my shopify dev store. 

## Expected behavior
The steps should allow the app to be installed in the dev store

## Actual behavior
Cannot get past Oauth error screens (see steps to reproduce) 

## Steps to reproduce the problem

From target directory I run: 
npm init @shopify/app@latest -- --template php

cd <appfoldername>/web

composer install 

cp .env.example .env

touch storage/db.sqlite

Update .env file with full DB path 

php artisan key:generate

php artisan migrate

cd ..

npm run dev
	Yes, create it as a new app
	name the app with same name as the project

Follow the provided grok URL and get the "You are about to visit ... grok warning page and click visit site to get this error: 
<details>Oauth error invalid_request: Could not find Shopify API application with api_key_not_defined </details>
<img width="468" alt="image" src="https://user-images.githubusercontent.com/913906/212481532-167422ba-7da1-4723-92fe-fde8881b8eee.png">

There doesn't appear to be any additional documentation after this step, so I'm winging it from here:
I go into my Shopify partners account and add the client ID and client secret into the .env as SHOPIFY_API_KEY and SHOPIFY_API_SECRET variables. 

** note I also ran 'npm run shopify env show' to verify which values should be added. 

re-run npm run dev 
Selected Always by default to have Shopify automatically update app's URL for preview experience
follow the new ngrok URL and now get this error:  <details>Oauth error invalid_scope: The access scope is invalid: not_defined</details>
<img width="468" alt="image" src="https://user-images.githubusercontent.com/913906/212481695-a1b23a60-c0f3-4e69-95cd-07be30dbfa01.png">

Now I have tried to get past this a couple ways: 
I've logged into my dev store and created an app there with the same name, and set an initial scope and added that to the .env  and I have tried simply adding the SCOPES variable with an initial value to the .env file (out of concern that creating an app with the same name in the dev store might end up with a different client id etc.), both paths ended with the same result. 

I try npm run dev again and this time get <details>Oauth error invalid_request: The redirect_uri is not whitelisted</details>. 
<img width="468" alt="image" src="https://user-images.githubusercontent.com/913906/212482059-e02baec6-584f-44ce-8252-dfe7bd33b66f.png">

I go back into my Partners account and access the App setup and validate the URLs: 

App URL: https://random-number.ngrok.io/

Allowed redirectionURL(s) 
https://random-number.ngrok.io/auth/callback
https://random-number.ngrok.io/auth/shopify/callback
https://random-number.ngrok.io/api/auth/callback

I have tried removing the one with Shopify in the path, as that is something that has been referenced in other posts around the same whitelist error. I have also tried setting the auto update of the URLs in the app to never and manually entered them, I've tried only using the auth/callback redirect, all with the same result. I simply have not been able to get past this error and actually install the app in my dev store to work with it. 

Any insight into how to get past these errors successfully would be appreciated.  I have not gotten any responses or support from the Shopify community boards and have not found success via stack overflow other searches etc. I would very much like to use Laravel for this custom app as I am far more familiar with it than some of the other app options.

**Comment by rafaelstz** [comment]
Hi @MidwestInfoDev, check your `.env.` file to make sure you have your **HOST** and **DB_DATABASE** variables updated, the DB must be the entire path and the HOST can be updated as soon as you run the app.
This is an example:

```
APP_NAME="Rafael Test"
APP_ENV=local
APP_KEY=base64:Pkghasdkgasdfasdfasfadsf=
APP_DEBUG=true

API_VERSION=2023-01
SHOPIFY_API_KEY=1836478163481267348
SHOPIFY_API_SECRET=ct1384709123840981234
SCOPES=read_customers,read_products,write_products,read_draft_orders,write_draft_orders
HOST=5132-4321-fa49-1344-2e00-8cc7-70e9-9abd-9854.ngrok.io

LOG_CHANNEL=stack
LOG_LEVEL=debug

DB_CONNECTION=sqlite
# The path to the database file should be absolute, make sure to update it!
DB_DATABASE=/Users/rafael/shopifyapps/myapphere/web/storage/db.sqlite
DB_FOREIGN_KEYS=true

```

**Comment by MidwestInfoDev** [comment]
I believe I may have found the issue. I was able to access the app via localhost and saw the Laravel error that was actually causing the issue:
<img width="468" alt="image" src="https://user-images.githubusercontent.com/913906/212777232-6590de00-c528-4d2f-a33d-789f58b4c1fa.png">

I saw from your response @rafaelstz that you included a HOST variable in your .env file. If I take the additional step of adding the newly generated ngrok URI to the .env file BEFORE I follow the URL in a browser, it works and I am able to reach the install approval screen in my dev store. 

Probably need to do a pull request to update the README for the repo. Thanks @rafaelstz  for helping to lead me to the answer!
